courses/models.py

```diff
@@ -250,3 +253,25 @@ def is_unexpired(self):
         Checks if the course is not expired
         """
         return not self.is_past and self.is_upgradable and self.is_not_beyond_enrollment
+
+    @property
+    def can_freeze_grades(self):
+        """
+        Checks if the final grades can be frozen.
+        """
+        if self.freeze_grade_date is None:
+            raise ImproperlyConfigured('Missing freeze_grade_date')
+        return datetime.now(pytz.utc) > self.freeze_grade_date
+
+    @classmethod
+    def get_freezable(cls):
+        """
+        Returns a queryset of all the runs that can freeze final grade according to the freeze date.
+        """
+        course_runs = cls.objects.exclude(
+            freeze_grade_date=None
```

Comment:
Is there ever a case where `CourseRun` should have `freeze_grade_date` == `None`, except in the case where it's not set yet?